### DISCUSSION TITLE
How to remove nodeSelector using pod_overrride?

### DISCUSSION THREAD

**Discussion by shisoderohit23** [apache/airflow#25339] [Q&A]
I am deploying airflow on GKE and for cost saving purposes I have all workers running on [spot VM](https://cloud.google.com/kubernetes-engine/docs/how-to/autopilot-spot-pods) but there are few that I do want to run on standard VM so that they finish ASAP.

To accomplish this I need to remove the `nodeSelector` I have on worker pod, and I was trying to use `executor_config` `pod_override` to set it to None from the one set as default for spot VM - `"cloud.google.com/gke-spot": "true"`

My code is as follows for one sample operator - 
```
BashOperator(
        task_id="echo_task",
        bash_command="sleep 30 && echo \"Hello ${RUN_NAME}\" ",
        executor_config={
            "pod_override": k8s.V1Pod(
                spec=k8s.V1PodSpec(
                    containers=[
                        k8s.V1Container(
                            name="base",
                            env=[
                                k8s.V1EnvVar(name="RUN_NAME", value="Rohit")
                            ],
                        )
                    ],
                    node_selector=None
                )
            )
        }
```

But this is not working. Pods are still having the default nodeSelector and I tried setting it to a blank dict like - `nodeSelector={}` but that did not work as well

**Comment by potiuk** [comment]
I vaguely recall discussion about it @jedcunningham @ephraimbuddy  - but I am not sure what was the output :)
